Bug marisor board

RDaniel99 opened the pull request.

Ciornache committed:
added ticket 0A + ticket 0B

Ciornache committed:
Changed half of the project

RDaniel99 committed:
Mob programming together

Ciornache committed:
added ticket(#7) + ticket(#8) + To Do's

Ciornache committed:
solved issues

Ciornache committed:
solved little bug

Ciornache committed:
Merge pull request #40 from Junior-Coders-Devs/ticket-#2,#7,#8

Ticket #2,#7,#8

RDaniel99 committed:
mob programming

Ciornache committed:
solved the bug hopefully

RDaniel99 committed:
Added singleton to Board

RDaniel99 committed:
Refactor player hierarchy

RDaniel99 committed:
Refactor movevalidator and added ticket 5

Ciornache commented on Chess/knight.cpp:
```diff
@@ -1,6 +1,17 @@
+
+    MoveBy m1(1, 2), m2(1, -2), m3(-1, 2), m4(-1, -2), m5(2, 1), m6(2, -1), m7(-2, 1), m8(-2, -1);
+
+    std::vector<MoveBy> knightMoves{m1, m2, m3, m4, m5, m6, m7, m8};
```
Poate putem extinde asta si la celelalte piese dar nu stiu daca are sens. 
Am putea la getPositions la fiecare piesa sa avem acum vector<MoveBy> nu vector<position>

Andreibatman replied:
Nu stiu cat de folositoare e ideea asta, mai ales la regina doar n-o sa punem toate cazurile posibile, mai ales ca la ea conteaza si unde e pe tabla

Ciornache commented on Chess/checkvalidator.cpp:
```diff
@@ -3,16 +3,12 @@
+void CheckValidator::isCheck(Color player)
 {
-    /*
-        Spor
```
urat

Ciornache commented on Chess/humanplayer.cpp:
```diff
@@ -1,61 +1,60 @@
 
 
-void HumanPlayer::waitForClick(int kind, int &row, int &column)
+position HumanPlayer::waitForClick()
```
n-ar trb aici sa asteptam doar click-ul?
poate putem face o functie separata care returneaza position convertToPosition sau ceva la stilu

RDaniel99 replied:
Super ok commentul.
Pe moment am redenumit din `waitForClick` in `getClick` ca sa se inteleaga ca nu doar asteptam
Un lucru foarte important la naming e ca o functie sa nu faca mai mult decat spune :D GG!

RDaniel99 replied:
`getPositionClicked` ramane

Andreibatman committed:
Quality of life changes

Schimbat din positions in moveby + bishop validpositions + jumatate din task-u cu check

RDaniel99 committed:
CR addressed

RDaniel99 committed:
CR Addressed

Andreibatman reviewed (approved):
Ggs
